cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -251,17 +265,37 @@ class SimpleArrayMixinCalculators
 
     value_type median() const
     {
-        auto athis = static_cast<A const *>(this);
-        const size_t n = athis->size();
-        small_vector<T> acopy(n);
-        auto sidx = athis->first_sidx();
+        MODMESH_TIME("median()")
+        A const * athis = nullptr;
+        size_t n = 0;
+        small_vector<T> acopy;
         size_t i = 0;
-        do
+        shape_type sidx;
+        value_type median_result = 0;
+
         {
-            acopy[i] = athis->at(sidx);
-            ++i;
-        } while (athis->next_sidx(sidx));
-        return median_op(acopy);
+            MODMESH_TIME("median::init()")
+            athis = static_cast<A const *>(this);
+            n = athis->size();
+            acopy = small_vector<T>(n);
+            sidx = athis->first_sidx();
+        }
+
+        {
+            MODMESH_TIME("median::copy()")
```

Comment:
It is OK to use a profiling node for this section which should take time.  But it will be better to factor it out to a helper function, if you really want to profile it.

I don't think you need to profile it, though.